sessionctx/stmtctx/stmtctx.go

```diff
@@ -229,28 +232,40 @@ func (sc *StatementContext) ResetStmtCache() {
 
 // SQLDigest gets normalized and digest for provided sql.
 // it will cache result after first calling.
-func (sc *StatementContext) SQLDigest() (normalized, sqlDigest string) {
+func (sc *StatementContext) SQLDigest() (normalized string, sqlDigest *parser.Digest) {
 	sc.digestMemo.Do(func() {
 		sc.digestMemo.normalized, sc.digestMemo.digest = parser.NormalizeDigest(sc.OriginalSQL)
 	})
 	return sc.digestMemo.normalized, sc.digestMemo.digest
 }
 
 // InitSQLDigest sets the normalized and digest for sql.
-func (sc *StatementContext) InitSQLDigest(normalized, digest string) {
+func (sc *StatementContext) InitSQLDigest(normalized string, digest *parser.Digest) {
 	sc.digestMemo.Do(func() {
 		sc.digestMemo.normalized, sc.digestMemo.digest = normalized, digest
 	})
 }
 
 // GetPlanDigest gets the normalized plan and plan digest.
-func (sc *StatementContext) GetPlanDigest() (normalized, planDigest string) {
+func (sc *StatementContext) GetPlanDigest() (normalized string, planDigest *parser.Digest) {
 	return sc.planNormalized, sc.planDigest
 }
 
+// GetResourceGroupTag gets the resource group of the statement.
+func (sc *StatementContext) GetResourceGroupTag() []byte {
+	if len(sc.resourceGroupTag) > 0 {
+		return sc.resourceGroupTag
+	}
+	_, sqlDigest := sc.SQLDigest()
+	sc.resourceGroupTag = resourcegrouptag.EncodeResourceGroupTag(sqlDigest, sc.planDigest)
```

Comment:
could we just marshal once for an stmt? it seems N kv requests in a SQL should be same resourceTagValue

Reply:
Currently, already cache the marshal result in `sc.resourceGroupTag`. Normally it should only marshal once.